src/detection_win.cpp

```diff
@@ -203,49 +214,21 @@ void EIO_Find(uv_work_t* req) {
 /**********************************
  * Local Functions
  **********************************/
-void ToUpper(char * buf) {
-	char* c = buf;
-	while (*c != '\0') {
-		*c = toupper((unsigned char)*c);
-		c++;
-	}
-}
-
-
-void extractVidPid(char * buf, ListResultItem_t * item) {
-	if(buf == NULL) {
-		return;
-	}
-
-	ToUpper(buf);
-
-	char* string;
-	char* temp;
-	char* pidStr, *vidStr;
-	int vid = 0;
-	int pid = 0;
 
-	string = new char[strlen(buf) + 1];
-	memcpy(string, buf, strlen(buf) + 1);
-
-	vidStr = strstr(string, VID_TAG);
-	pidStr = strstr(string, PID_TAG);
-
-	if(vidStr != NULL) {
-		temp = (char*) (vidStr + strlen(VID_TAG));
-		temp[4] = '\0';
-		vid = strtol (temp, NULL, 16);
-	}
+void ExtractDetails(TCHAR* devicePath, ListResultItem_t* item) {
+	// expected format:
+	// \\?\USB#Vid_04e8&Pid_503b#0002F9A9828E0F06#{a5dcbf10-6530-11d2-901f-00c04fb951ed}
+	CString dpcs = devicePath + 4;
```

Comment:
Unused?

What does `dpcs` stand for? *device path CString*

Reply:
This can certainly be removed. It was unintentionally left over from a previous implementation before I decided to use the Left/Mid methods.